### DISCUSSION TITLE
How to disable auto populate when you define OneToOne relation ?

### DISCUSSION THREAD

**Discussion by huyhoang160593** [mikro-orm/mikro-orm#5028] [Q&A] [answered]
Like the question say, I only need the query findOne to return the reference id, but instead they always generate the sql with join query and return the whole entities, especially with OneToOne relation, Is there a way to turn off that? Seem like define populate: false in the seconds argument is doesn't do anything
NodeJS: 20.10.0
Mikro-orm: 5.9.5

**Comment by B4nan** [comment, marked as answer]
maybe try the search in the docs, its not that hard :]

https://mikro-orm.io/docs/next/configuration#auto-join-of-11-owners

**Reply by huyhoang160593** [reply]
I have tried to find them first in the discussion and issues and also the docs with the keyword 'auto populate' for the whole days :(. Thanks you for pointing me to the right path I need to see

**Reply by B4nan** [reply]
because it's not population, you don't get a populated relation, just a virtual FK so you can tell whether the value is there or not.
